File: benchmarks/python/02_fibonacci_iterative.py
import time
import math

def fib_iter(n):
    a, b = 0, 1
    for _ in range(n):
        a, b = b, a + b
    return a

# Warmup
fib_iter(1000)

n = 50
iterations = 1000000

start = time.perf_counter()
result = 0
for _ in range(iterations):
    result = fib_iter(n)
elapsed = (time.perf_counter() - start) * 1000

ops_per_sec = math.floor(iterations / (elapsed / 1000))
print("BENCHMARK: fibonacci_iterative")
print(f"n={n}, iterations={iterations}")
print(f"result={result}")
print(f"time={elapsed}ms")
print(f"ops_per_sec={ops_per_sec}")

File: benchmarks/go/02_fibonacci_iterative.go
package main

import (
	"fmt"
	"math"
	"time"
)

func fibIter(n int) int {
	a, b := 0, 1
	for i := 0; i < n; i++ {
		a, b = b, a+b
	}
	return a
}

func main() {
	fibIter(1000)

	n := 50
	iterations := 1000000

	start := time.Now()
	var result int
	for i := 0; i < iterations; i++ {
		result = fibIter(n)
	}
	elapsed := time.Since(start).Seconds() * 1000

	opsPerSec := math.Floor(float64(iterations) / (elapsed / 1000))

	fmt.Println("BENCHMARK: fibonacci_iterative")
	fmt.Printf("n=%d, iterations=%d\n", n, iterations)
	fmt.Printf("result=%d\n", result)
	fmt.Printf("time=%.6fms\n", elapsed)
	fmt.Printf("ops_per_sec=%.0f\n", opsPerSec)
}
